frontend/src/static/js/v_summary.js

```diff
@@ -323,42 +377,76 @@ window.vSummary = {
-        const mergedFileTypeContribution = {};
-        let mergedVariance = 0;
-        let totalMergedCommits = 0;
-        let totalMergedCheckedFileTypeCommits = 0;
+    mergeGroupByIndex(filtered, groupIndex) {
+      const dateToIndexMap = {};
+      const mergedCommits = [];
+      const mergedFileTypeContribution = {};
+      let mergedVariance = 0;
+      let totalMergedCommits = 0;
+      let totalMergedCheckedFileTypeCommits = 0;
 
-        group.forEach((user) => {
-          this.mergeCommits(user, mergedCommits, dateToIndexMap);
+      filtered[groupIndex].forEach((user) => {
+        this.mergeCommits(user, mergedCommits, dateToIndexMap);
 
-          this.mergeFileTypeContribution(user, mergedFileTypeContribution);
+        this.mergeFileTypeContribution(user, mergedFileTypeContribution);
 
-          totalMergedCommits += user.totalCommits;
-          totalMergedCheckedFileTypeCommits += user.checkedFileTypeContribution;
-          mergedVariance += user.variance;
-        });
+        totalMergedCommits += user.totalCommits;
+        totalMergedCheckedFileTypeCommits += user.checkedFileTypeContribution;
+        mergedVariance += user.variance;
+      });
+
+      mergedCommits.sort(window.comparator((ele) => ele.date));
+      filtered[groupIndex][0].commits = mergedCommits;
+      filtered[groupIndex][0].fileTypeContribution = mergedFileTypeContribution;
+      filtered[groupIndex][0].totalCommits = totalMergedCommits;
+      filtered[groupIndex][0].variance = mergedVariance;
+      filtered[groupIndex][0].checkedFileTypeContribution = totalMergedCheckedFileTypeCommits;
+
+      filtered[groupIndex] = filtered[groupIndex].slice(0, 1);
+    },
 
-        mergedCommits.sort(window.comparator((ele) => ele.date));
-        group[0].commits = mergedCommits;
-        group[0].fileTypeContribution = mergedFileTypeContribution;
-        group[0].totalCommits = totalMergedCommits;
-        group[0].variance = mergedVariance;
-        group[0].checkedFileTypeContribution = totalMergedCheckedFileTypeCommits;
+    hasGroupMerged() {
+      if (Object.keys(this.mergedGroupsMap).length === 0) {
+        return true;
+      }
+      return Object.keys(this.mergedGroupsMap)
+          .map((key) => this.mergedGroupsMap[key])
+          .includes(true);
+    },
+
+    isAllGroupsMerged() {
+      if (Object.keys(this.mergedGroupsMap).length === 0) {
+        return false;
+      }
+      return !Object.keys(this.mergedGroupsMap)
+          .map((key) => this.mergedGroupsMap[key])
+          .includes(false);
+    },
 
-        // clear all users and add merged group in filtered group
-        filtered[groupIndex] = [];
-        filtered[groupIndex].push(group[0]);
+    resetMergedGroupsMap() {
+      const mergedGroupsMap = {};
+      this.filtered.forEach((group) => {
+        mergedGroupsMap[this.getGroupName(group)] = false;
       });
+      this.mergedGroupsMap = mergedGroupsMap;
+    },
+
+    handleMergeGroup(groupName) {
```

Comment:
`handleMergeGroup` sounds a bit vague when the logic is only setting the merged property to true. Perhaps `setGroupToMerge` may be a better name?

Reply:
The reason I use `handleXXX` is to signify that the function is specifically used for UI event handler. It just happens to be that the way which it handles the event is only setting the property to true.

Reply:
Perhaps another way to make it more specific is `handleMergeGroupIconClicked` but that's a bit too long?